spring-cloud-stream/src/main/java/org/springframework/cloud/stream/error/AbstractMessageChannelErrorConfigurer.java

```diff
@@ -0,0 +1,273 @@
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.MessageHandler;
+import org.springframework.messaging.SubscribableChannel;
+import org.springframework.retry.backoff.ExponentialBackOffPolicy;
+import org.springframework.retry.policy.SimpleRetryPolicy;
+import org.springframework.retry.support.RetryTemplate;
+import org.springframework.util.Assert;
+
+/**
+ * Base implementation of {@link BinderErrorConfigurer}, provides default implementations for
+ * {@link Binder} implementations based on {@link MessageChannel} abstractions.
+ * Binder implementations are still responsible to subclass this and provide a configure implementation
+ * that fits their specific needs.
+ * @author Vinicius Carvalho
+ * @author Gary Russel
+ */
+public abstract class AbstractMessageChannelErrorConfigurer<C extends ConsumerProperties> implements BinderErrorConfigurer<C,MessageChannel>, ApplicationContextAware{
+
+	private volatile AbstractApplicationContext applicationContext;
+
+	private Map<String,ErrorInfrastructure> errorInfrastructureMap = new ConcurrentHashMap<>();
+
+	protected final Log logger = LogFactory.getLog(getClass());
+
+	protected AbstractApplicationContext getApplicationContext() {
+		return this.applicationContext;
+	}
+
+	@Override
+	public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+		Assert.isInstanceOf(AbstractApplicationContext.class, applicationContext);
+		this.applicationContext = (AbstractApplicationContext) applicationContext;
+	}
+
+	@Override
+	public void register(Binding<MessageChannel> binding, String group, C consumerProperties) {
+		MessageProducerBinding consumerBinding = (MessageProducerBinding)binding;
+		ErrorMessageStrategy errorMessageStrategy = getErrorMessageStrategy();
+		ConfigurableListableBeanFactory beanFactory = getApplicationContext().getBeanFactory();
+		String errorChannelName = errorsBaseName(consumerBinding.getDestination(), group, consumerProperties);
+		SubscribableChannel errorChannel = null;
+		if (getApplicationContext().containsBean(errorChannelName)) {
+			Object errorChannelObject = getApplicationContext().getBean(errorChannelName);
+			if (!(errorChannelObject instanceof SubscribableChannel)) {
+				throw new IllegalStateException(
+						"Error channel '" + errorChannelName + "' must be a SubscribableChannel");
+			}
+			errorChannel = (SubscribableChannel) errorChannelObject;
+		}
+		else {
+			errorChannel = new BinderErrorChannel();
+			beanFactory.registerSingleton(errorChannelName, errorChannel);
+			errorChannel = (LastSubscriberAwareChannel) beanFactory.initializeBean(errorChannel, errorChannelName);
+		}
+		ErrorMessageSendingRecoverer recoverer;
+		if (errorMessageStrategy == null) {
+			recoverer = new ErrorMessageSendingRecoverer(errorChannel);
+		}
+		else {
+			recoverer = new ErrorMessageSendingRecoverer(errorChannel, errorMessageStrategy);
+		}
```

Comment:
IMHO this is really an SI issue. I would address it in ```ErrorMessageSendingRecoverer```. It already supports default error message strategy via one of its constructor
```
public ErrorMessageSendingRecoverer(MessageChannel channel) {
	this(channel, new DefaultErrorMessageStrategy());
}
```
. . . so consider modifying both constructors in ```ErrorMessageSendingRecoverer``` to something like this:
```
public ErrorMessageSendingRecoverer(MessageChannel channel) {
	this(channel, null);
}

public ErrorMessageSendingRecoverer(MessageChannel channel, ErrorMessageStrategy errorMessageStrategy) {
	setChannel(channel);
	setErrorMessageStrategy(errorMessageStrategy == null ? new DefaultErrorMessageStrategy() : errorMessageStrategy);
}
```